boards/arm/arduino_opta_m4/doc/index.rst

```diff
@@ -0,0 +1,161 @@
+.. _arduino_opta_m4_board:
+
+Arduino OPTA M4-Core
+#######################
+
+Overview
+********
+
+Opta™ is a secure micro Programmable Logic Controller (PLC)
+with Industrial Internet of Things (IoT) capabilities.
+
+Developed in partnership with Finder®, this device supports both the Arduino
+programming language and standard IEC-61131-3 PLC programming languages,
+such as Ladder Diagram (LD), Sequential Function Chart (SFC),
+Function Block Diagram (FBD), Structured Text (ST), and Instruction List (IL),
+making it an ideal device for automation engineers.
+
+For Zephyr RTOS, only the M4 is supported for now, making the M7 to run the PLC
+tasks while the M4 core under Zephyr acts as a coprocessor.
+
+Additionally, the device features:
+
+- Ethernet compliant with IEEE802.3-2002
+- 256-Mbit SDRAM
+- 4 x green color status LEDs
+- 1 x user and reset push-button
+- 10 x analog inputs
+- 4 x isolated relay outputs
+
+.. image:: img/arduino_opta.jpeg
+     :align: center
+     :alt: ARDUINO-OPTA
+
+More information about the board can be found at the `ARDUINO-OPTA website`_.
+More information about STM32H747XIH6 can be found here:
+
+- `STM32H747XI on www.st.com`_
+- `STM32H747xx reference manual`_
+- `STM32H747xx datasheet`_
+
+Supported Features
+==================
+
+The current Zephyr arduino_opta_m4 board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| IPM       | on-chip    | virtual mailbox based on HSEM       |
```

Comment:
prob should be dropped as well?

Reply:
This is something I find ambiguous sometimes, because although both SPI and UART are not in the DTS file, saying they are not supported is not entirely true, since just an overlay fragment makes it to be usable, so put on the boarrd document that other features not mentioned are not supported may discourage users to try to use other features that are actually well supported by the SoC that is not mentioned in its board documentation, so I'm wondering if  we can put on this table two extra columns called supported by SoC, supported ready on the Board.

Anyway, it is just a personal opinion, for now I will remove both UART and SPI :)